server/models/services/AtService.js

```diff
@@ -553,6 +553,40 @@ const getUniqueAtVersionsForReport = async testPlanReportId => {
     return results;
 };
 
+/**
+ * @param {object} params - values to be used to create or find the AtVersion record
+ * @param {string[]} atVersionAttributes  - AtVersion attributes to be returned in the result
+ * @param {string[]} atAttributes  - At attributes to be returned in the result
+ * @param {object} options - Generic options for Sequelize
+ * @param {*} options.transaction - Sequelize transaction
+ * @returns {BrowserVersion}
+ */
+const findOrCreateAtVersion = async (
+    { atId, name, releasedAt },
+    atVersionAttributes = AT_VERSION_ATTRIBUTES,
+    atAttributes = AT_ATTRIBUTES,
+    options = {}
+) => {
+    let version = await getAtVersionByQuery(
+        { atId, name },
+        atVersionAttributes,
+        atAttributes,
+        options
+    );
+
+    if (!version) {
+        /* TODO: releasedAt manually entered by users submitting new version, support actual dates in automation */
+        version = await createAtVersion(
+            { atId, name, releasedAt: releasedAt ?? new Date(Date.now()) },
```

Comment:
```suggestion
            { atId, name, releasedAt: releasedAt ?? new Date() },
```

Reply:
Oh nice! Didn't know that the constructor would fallback to Date.now()